src/Dapr.Client/DaprSubscribeConfigurationSource.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Grpc.Core;
+using Autogenerated = Dapr.Client.Autogen.Grpc.v1;
+
+namespace Dapr.Client
+{
+    internal class DaprSubscribeConfigurationSource : ConfigurationSource
+    {
+        private AsyncServerStreamingCall<Autogenerated.SubscribeConfigurationResponse> call;
+        private string id = string.Empty;
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="sourceStreamRequest">The request that is used to start the configuration streaming with Dapr.</param>
+        /// <param name="internalClient">The internal client used to call Dapr and begin a gRPC stream.</param>
+        /// <param name="callOptions">The options for the call to Dapr.</param>
+        internal DaprSubscribeConfigurationSource(
+            Autogenerated.Dapr.DaprClient internalClient,
+            Autogenerated.SubscribeConfigurationRequest sourceStreamRequest,
+            CallOptions callOptions) : base()
+        {
+            this.call = internalClient.SubscribeConfigurationAlpha1(sourceStreamRequest, callOptions);
+        }
+
+        /// <inheritdoc/>
+        public override string Id => id;
+
+        public override IAsyncEnumerator<IEnumerable<ConfigurationItem>> GetAsyncEnumerator(CancellationToken cancellationToken = default)
+        {
+            return new DaprSubscribeConfigurationEnumerator(call, streamId => setIdIfNullOrEmpty(streamId), cancellationToken);
+        }
+
+        private void setIdIfNullOrEmpty(string id)
+        {
+            if (string.IsNullOrEmpty(this.id))
+            {
+                this.id = id;
+            }
+        }
```

Comment:
So is there no way to get the ID without enumerating the items? This design seems broken.

- What about cases where my subscribe doesn't match any existing keys? 
- What about cases where I subscribe, but then have to shut down before reading any keys?

Reply:
Not currently, no. You've already commented on the issue where I ask for it in the dapr API but for anyone else who's interested: https://github.com/dapr/dapr/issues/4546